### DISCUSSION TITLE
Customizing the login-reset-password page

### DISCUSSION THREAD

**Discussion by chris-diam** [keycloakify/keycloakify#246] [Q&A]
Hello! I have a question, I know this may be stupid but I am struggling when it comes to customizing the login-reset-password and generally how to change the messages e.g. the "Forgot Password" header as depicted below.
Thanks in advance!

![image](https://user-images.githubusercontent.com/91747939/217534252-5799cb10-1977-4a28-8592-8d47f45e3c8c.png)

**Comment by garronej** [comment]
Hi @chris-diam,  

We can see that the message in question is `"emailForgotTitle"`

<img width="1002" alt="image" src="https://user-images.githubusercontent.com/6702424/217535183-cce6dbda-542f-40e6-ba5b-b671a6df3b0e.png">  

<img width="1090" alt="image" src="https://user-images.githubusercontent.com/6702424/217536035-0e7ef99a-9a53-4b6d-b247-6592b8bdba0f.png">




You can use the `i18n` API to define new messages or overwrite existing ones.  
https://github.com/garronej/keycloakify-advanced-starter/blob/main/src/KcApp/i18n.ts  

https://docs.keycloakify.dev/adding-text-keys   

Good luck

**Reply by atefBB** [reply]
How can customize the whole page looking ? coping it from `src/lib/components` works ?

**Reply by garronej** [reply]
You can copy pass [this](https://github.com/keycloakify/keycloakify/blob/main/src/login/pages/LoginResetPassword.tsx) to your project yes.  

But you can also just run `npx eject-keycloak-page` and select `login-reset-password.ftl`

**Reply by atefBB** [reply]
Thanks for quick response ! This what I've do !
